### PR TITLE
SIG-3575 Document OpenID Providers (OIDC) api

### PR DESCRIPTION
**Pull request by evoronkin**

### CODE REVIEW

**Review by generateui — COMMENTED** [review]
The commit message is incorrect since it mentions OpenId, where I understand it is about OpenId Connect which is *distinctly different*. Thus must absoltely be fixed. Please also check all other PRs, confluence pages, code changes (your notes?) etc if such mistakes are made in naming.

File: _data/api-swagger.yaml [hunk 1]
```diff
@@ -1053,6 +1054,28 @@ definitions:
       For example, in the Netherlands, personal number (BSN) can not be used for privacy reasons
       and is substituted by document number such as passport number, but in other countries this poses no problem.
 
+  OpenID Providers:
+    allOf:
+    - $ref: '#/definitions/Verification'
+    - type: object
+      description: |
+        Adds an OpenID verification step to the signing process.
```
Comment (generateui) [inline _data/api-swagger.yaml:1062]: rename to OpenId Connect

File: _data/api-swagger.yaml [hunk 2]
```diff
@@ -1053,6 +1054,28 @@ definitions:
...
+      properties:
+        ProviderName:
+          type: string
+          description: |
+            (OPTIONAL) When value is provided, it should match the configured one for the organization.
+            If the value does not match then the full list of the organization providers will be given for the end-user to select during the verification step.
+          default: null
+    example: |
+      A OpenID verification response may have the following content:
```
Comment (generateui) [inline _data/api-swagger.yaml:1071]: An*

rename OpenID to OpenId Connect

File: _data/api-swagger.yaml [hunk 3]
```diff
@@ -1053,6 +1054,28 @@ definitions:
...
+      ```json
+      {
+          "Type": "OpenID Providers",
```
Comment (generateui) [inline _data/api-swagger.yaml:1074]: "Type": "OpenID Providers",

This should/must be `OpenId Connect Providers`

File: _data/api-swagger.yaml [hunk 4]
```diff
@@ -1053,6 +1054,28 @@ definitions:
...
+    allOf:
+    - $ref: '#/definitions/Verification'
+    - type: object
+      description: |
+        Adds an OpenID verification step to the signing process.
+      properties:
+        ProviderName:
+          type: string
+          description: |
+            (OPTIONAL) When value is provided, it should match the configured one for the organization.
```
Comment (generateui) [inline _data/api-swagger.yaml:1067]: explicitly refer to "the configured one"

File: _data/api-swagger.yaml (hunk 1 above)
Comment (PaulLoth) [reply _data/api-swagger.yaml:1062]: In our api we call it OpenID Providers. Our friendly name is openID, so good to have one term for both api, portal and documentation.

File: _data/api-swagger.yaml (hunk 3 above)
Comment (PaulLoth) [reply _data/api-swagger.yaml:1074]: This is documentation for an api call that is named OpenID Providers. So changing in documentation will be false and not functional. We named it OpenID Providers.

File: _data/api-swagger.yaml (hunk 4 above)
Comment (evoronkin) [reply _data/api-swagger.yaml:1067]: Provider Names is configured in a free flexible string way, we do not have hard coded names. So what do you refer here to be explicit?

**Comment by evoronkin** [comment]
> The commit message is incorrect since it mentions OpenId, where I understand it is about OpenId Connect which is _distinctly different_. Thus must absoltely be fixed. Please also check all other PRs, confluence pages, code changes (your notes?) etc if such mistakes are made in naming.

It was already agreed with the product to call:
Type: "OpenID Providers", defined also in the standard as commonly used to refer to the OIDC;
OpenID is the common understandable ID which should be validated, users will understand this and they do not have to know about OIDC protocol.

**Review by BerendJonkman — APPROVED** [review]
I do agree with Ruud on that OpenID Connect would have been more correct since there is also a standard called OpenID. However since this is a decision by Product I will approve this PR.
Approved for tech review - `b78286c404578365f3ac616ceb57ab870864dcef`

**Comment by ATimmeh33** [comment]
Approved tech review - b78286c404578365f3ac616ceb57ab870864dcef

**Pull request merged**